Automattic/vip-design-system
# Show the selected value in the input of Form.Autocomplete

## Description

Right now if you start writing in the select and the focus away from it, the old value is still selected and you have no chance of knowing that. 
With this PR I try to introduce a new behaviour where if the select loses focus and no selection has been made, then the select is resetted. 

This might not be the best solution, but it will allow us to fix some confusion while using the autocomplete.

I plan to also fix the original component and remove this. 

KNOWN BUG: If you input some data and you use the keyboard to move and make a selection, the input field will be resetted to either the selected value or empty if no selection  has been  made. 

### Before
![Before Screenshot screencast 2023-01-25 15-43-55](https://user-images.githubusercontent.com/668251/214600985-db279318-97a6-470b-b782-c84266be04b7.gif)

### After

![After Screenshot screencast 2023-01-25 16-09-58](https://user-images.githubusercontent.com/668251/214601081-580d6a70-b6ca-4dec-a55e-985618889cf6.gif)

**This PR also introduces support for controls in Storybook for the Autocomplete component.**
This way you can test the `resetOnBlur` filter for every example (the only one not supported at the moment is the Debounce)
![CleanShot 2023-02-10 at 18 45 18@2x](https://user-images.githubusercontent.com/668251/218160434-5db0ae85-df9b-4aeb-9b5a-fabc291ca96f.png)


## Checklist


## Steps to Test

Outline the steps to test and verify the PR here.

1. Pull down PR.
1. `npm run dev`.
1. Open Storybook for the component http://localhost:6006/?path=/story/form-autocomplete--default 
1. Try to select a field, then edit the input and move the focus away (clicking in the page or moving away with a tab). The field should reset to the selected state
1.After selecting a state, delete all the contents, the when losing focus it should keep the empty contents.
